### PR TITLE
Fix remaining julia 0.7 dep warnings

### PR DESCRIPTION
**Pull request by davidanthoff**
I only saw #157 after I had finished this PR, so it does that one as well. This PR here fixes more warnings, though, so we might as well just merge this here.

### CODE REVIEW

**Comment by davidanthoff** [comment]
I don't understand why this PR would break things on 0.6, it really doesn't seem to change anything there?

File: src/ssl.jl [hunk 1]
```diff
@@ -207,8 +207,8 @@ function handshake(ctx::SSLContext)
             try
                 wait(ctx.bio.readnotify)
             catch e
-                if e isa Base.UVError
-                    # Ignore read errors (UVError ECONNRESET)
+                if e isa Base.IOError
```
Comment (quinnj) [inline src/ssl.jl]: this will need to be compat-ified, right?

Comment (davidanthoff) [reply src/ssl.jl]: I think the whole larger block is already behind a version check.

**Comment by JeffBezanson** [comment]
Can this please be merged? Existing versions will still work on 0.6 so I don't see a major problem there.

**Comment by tkelman** [comment]
It needs to drop 0.6 in REQUIRE if it doesn't work there.

File: src/ssl.jl (hunk 1 above)
Comment (tkelman) [reply src/ssl.jl]: if the renaming happened after 0.7.0-alpha.0, then the cutoff should be changed

**Comment by JeffBezanson** [comment]
Yes, you're right about REQUIRE.

**Pull request merged**